### PR TITLE
Migrate tests l-n to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/test/memory_service_test.dart [hunk 1]
```diff
@@ -161,7 +161,7 @@ Future<void> collectSamples([int sampleCount = defaultSampleSize]) async {
 }
 
 void checkHeapStat(ClassHeapDetailStats classStat, String className,
-    {int instanceCount, int accumulatorCount}) {
+    {int? instanceCount, int? accumulatorCount}) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/memory_service_test.dart:164]: nit: tc

File: packages/devtools_app/test/navigation_test.dart [hunk 2]
```diff
@@ -37,7 +35,7 @@ void main() {
     testWidgets(
         'Generates a route name with parameters with an empty route in the context',
         (WidgetTester tester) async {
-      String generatedRoute;
+      late String generatedRoute;
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/navigation_test.dart:38]: why are some of these late and some are nullable? I would think they should all be late or all be nullable

File: packages/devtools_app/test/network_controller_test.dart [hunk 3]
```diff
@@ -95,7 +95,7 @@ void main() {
       const numRequests = 16;
 
       // Refresh network data and ensure requests are populated.
-      await controller.networkService.refreshNetworkData();
+      await controller.networkService!.refreshNetworkData();
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/network_controller_test.dart:98]: network service should not be nullable. See Ben's PR here: https://github.com/flutter/devtools/pull/3880/files#diff-5823db4661b7010348a2cbb6910f27b950c8cc3b63b21744bce8fe5c779b47d8

File: packages/devtools_app/test/network_table_test.dart [hunk 4]
```diff
@@ -139,7 +139,7 @@ void main() {
     FakeServiceManager fakeServiceManager;
     SocketProfile socketProfile;
     HttpProfile httpProfile;
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/network_table_test.dart:141]: do all these variables need to be late?

File: packages/devtools_app/test/memory_service_test.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/test/memory_service_test.dart:164]: done

File: packages/devtools_app/test/network_controller_test.dart (hunk 3 above)
Comment (polina-c) [reply packages/devtools_app/test/network_controller_test.dart:98]: done

File: packages/devtools_app/test/network_table_test.dart (hunk 4 above)
Comment (polina-c) [reply packages/devtools_app/test/network_table_test.dart:141]: yes

**Review by kenzieschmoll — APPROVED** [review]

**Pull request merged**